app/services/manage_action.rb

```diff
@@ -1,91 +1,85 @@
 # frozen_string_literal: true
 
 class ManageAction
-  attr_reader :params
-
   def self.create(params, extra_params: {}, skip_queue: false, skip_counter: false)
     new(params, extra_params: extra_params, skip_queue: skip_queue, skip_counter: skip_counter).create
   end
 
   def initialize(params, extra_params: {}, skip_queue: false, skip_counter: false)
-    @params = params
+    @params = params.clone
     @skip_queue = skip_queue
     @skip_counter = skip_counter
-    @extra_attrs = extra_params
+    @extra_attrs = extra_params.clone
```

Comment:
Would be great to write somewhere in a comment what these `extra_params` represent. There's at least a boolean `donation` key, is there anything else? The naming is pretty unclear about what these parameters are.